river/command/move.zig

```diff
@@ -0,0 +1,196 @@
+
+    const delta = try std.fmt.parseInt(i32, args[2], 10);
+    const direction = std.meta.stringToEnum(PhysicalDirection, args[1]) orelse
+        return Error.InvalidPhysicalDirection;
+
+    const view = get_view(seat) orelse return;
+    const output_box = get_output_dimensions(view);
+    const border_width = @intCast(i32, view.output.root.server.config.border_width);
+    switch (direction) {
+        .up => view.pending.box.y = std.math.clamp(
+            view.pending.box.y - delta,
+            border_width,
+            @intCast(i32, output_box.height - view.pending.box.height) - border_width,
+        ),
+        .down => view.pending.box.y = std.math.clamp(
+            view.pending.box.y + delta,
+            border_width,
+            @intCast(i32, output_box.height - view.pending.box.height) - border_width,
+        ),
+        .left => view.pending.box.x = std.math.clamp(
+            view.pending.box.x - delta,
+            border_width,
+            @intCast(i32, output_box.width - view.pending.box.width) - border_width,
+        ),
+        .right => view.pending.box.x = std.math.clamp(
+            view.pending.box.x + delta,
+            border_width,
+            @intCast(i32, output_box.width - view.pending.box.width) - border_width,
+        ),
+    }
+
+    apply(view);
+}
+
+pub fn snap(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len < 2) return Error.NotEnoughArguments;
+    if (args.len > 2) return Error.TooManyArguments;
+
+    const direction = std.meta.stringToEnum(PhysicalDirection, args[1]) orelse
+        return Error.InvalidPhysicalDirection;
+
+    const view = get_view(seat) orelse return;
+    const output_box = get_output_dimensions(view);
+    const border_width = @intCast(i32, view.output.root.server.config.border_width);
+    switch (direction) {
+        .up => view.pending.box.y = border_width,
+        .down => view.pending.box.y =
+            @intCast(i32, output_box.height - view.pending.box.height) - border_width,
+        .left => view.pending.box.x = border_width,
+        .right => view.pending.box.x =
+            @intCast(i32, output_box.width - view.pending.box.width) - border_width,
+    }
+
+    apply(view);
+}
+
+pub fn resize(
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len < 3) return Error.NotEnoughArguments;
+    if (args.len > 3) return Error.TooManyArguments;
+
+    const delta = try std.fmt.parseInt(i32, args[2], 10);
+    const orientation = std.meta.stringToEnum(Orientation, args[1]) orelse
+        return Error.InvalidOrientation;
+
+    const view = get_view(seat) orelse return;
+    const output_box = get_output_dimensions(view);
+    const border_width = @intCast(i32, view.output.root.server.config.border_width);
+    switch (orientation) {
```

Comment:
It'd be nice if we could unify these two code paths. Should be doable with pointers to `width/x` or `height/y`.